### DISCUSSION TITLE
Conditional globalAttribute?

### DISCUSSION THREAD

**Discussion by evanfuture** [ueberdosis/tiptap#1943] [General Questions]
Hi everyone.  I've been migrating from Quill to Tiptap, and in reproducing the same behaviour, I've had to build some of my own extensions.  It speaks to the power of tiptap that I can do so with ease.  One of the extensions I've built (upon the work of @sereneinserenade as shared [here](https://github.com/ueberdosis/tiptap/issues/1036) ), is an indent/outdent command.  You can see the extension here:

https://github.com/evanfuture/tiptaptop-extension-indent

My only issue with this is that it applies the node.attrs.indent property to *all* nodes, I guess because I'm using `addGlobalAttributes`.  I wonder if anyone knows a way to avoid setting that value for a node if the indent value is `null`?.

The reason this is important is because it adds `"attrs":{"indent":null}` to every single paragraph in the JSON, which can impact performance and balloon data-storage.

Potentially, I can write a custom parser in my project, but it would be nice to handle this directly within the extension.

Thanks in advance for any ideas/pointers!

**Comment by evanfuture** [comment]
@philippkuehn I guess you'd be the one to help point me in the right direction here?  It's enough of an issue that I might convert this into a proper Issue, but maybe there's already a better way to solve it?  (specifically, I want to only add the attribute when there is a value)

**Comment by philippkuehn** [comment]
Hey, no. Attributes are always stored in JSON. You can try to delete elements from the JSON afterwards. I'm not sure if that will cause any issues. But that won't happen in the core.

**Reply by evanfuture** [reply]
Ok.  I've implemented in my own codebase the manual removal of these.  So no rush here.  But I'm not sure I made myself clear:

Would you think of modifying the code of `packages/core/src/helpers/getAttributesFromExtensions.ts` to filter out attributes that don't have a default value?  Something like:
```
 globalAttributes.forEach(globalAttribute => {
      globalAttribute.types.forEach(type => {
        Object
          .entries(globalAttribute.attributes)
          .forEach(([name, currentAttribute]) => {
          const attribute = { ...defaultAttribute, ...currentAttribute };
           if (attribute.default !== null) {
              extensionAttributes.push({
                type,
                name,
                 attribute,
              })
            }
          })
      })
    })
```

My thought is that these globalAttributes should apply the default value if it's set, but if it isn't then they are being set to `null`, which doesn't seem all that useful?
